miri/lib.rs

```diff
@@ -287,6 +287,7 @@ impl<'mir, 'tcx: 'mir> Machine<'mir, 'tcx> for Evaluator<'tcx> {
             .interpret_interner
             .get_cached(cid.instance.def_id())
             .expect("uncached static");
+        // TODO: do a recursive copy
```

Comment:
No, no recursive copy. if you have

```rust
static FOO: &'static Foo = &BAR;
static BAR: Foo = Foo { field: &FOO };
```

Then BAR would end up referring to a copy of FOO instead of referring to it directly.

Reply:
We can't evaluate the static in the same EvalContext as far as I know, so we have to copy it anyway. A cache could be used just like `ecx.memory.data.mut_static` to prevent duplication.

Reply:
You can do it. Just push a stackframe for it as you had before https://github.com/solson/miri/pull/364/commits/907e67f6a36ae301caf4089ad53e7adb82f1f876

Reply:
But the `while ecx.step()? {}` will step out of the frame present at the beginning of the `init_static` function.

Reply:
ah, you can solve that by recording the stack size before pushing the frame (`self.frames().len()`)  in a variable and looping until you're back at the same frame.